include/rgb565.h

```diff
@@ -50,36 +58,41 @@ class Rgb16 {
 		pixel = static_cast<uint16_t>(r5 | g6 | b5);
 	}
 
+	Rgb888 ToRgb888() const
+	{
+		const auto r8 = Red5To8(pixel);
+		const auto g8 = Green6To8(pixel);
+		const auto b8 = Blue5To8(pixel);
+		return Rgb888(r8, g8, b8);
+	}
+
 	// Write to separate RGB 8-bit values
-	constexpr void ToRgb888(uint8_t& r8, uint8_t& g8, uint8_t& b8) const
+	void ToRgb888(uint8_t& r8, uint8_t& g8, uint8_t& b8) const
 	{
 		r8 = Red5To8(pixel);
 		g8 = Green6To8(pixel);
 		b8 = Blue5To8(pixel);
 	}
 
 	// Scoped conversion helper: RGB Red 5-bit to 8-bit
-	static constexpr uint8_t Red5To8(const uint16_t val)
+	static uint8_t Red5To8(const uint16_t val)
 	{
 		const auto red5 = (val & r5_mask) >> r5_offset;
-		const auto red8 = (red5 * 255 + 15) / 31;
-		return static_cast<uint8_t>(red8);
+		return static_cast<uint8_t>(rgb5_to_8_lut(red5));
 	}
 
 	// Scoped conversion helper: RGB Green 6-bit to 8-bit
-	static constexpr uint8_t Green6To8(const uint16_t val)
+	static uint8_t Green6To8(const uint16_t val)
 	{
-		const auto green5 = (val & g6_mask) >> g6_offset;
-		const auto green8 = (green5 * 255 + 31) / 63;
-		return static_cast<uint8_t>(green8);
+		const auto green6 = (val & g6_mask) >> g6_offset;
+		return static_cast<uint8_t>(rgb6_to_8_lut(green6));
 	}
 
 	// Scoped conversion helper: RGB Blue 5-bit to 8-bit
-	static constexpr uint8_t Blue5To8(const uint16_t val)
+	static uint8_t Blue5To8(const uint16_t val)
 	{
 		const auto blue5 = (val & b5_mask) >> b5_offset;
-		const auto blue8 = (blue5 * 255 + 15) / 31;
-		return static_cast<uint8_t>(blue8);
+		return static_cast<uint8_t>(rgb5_to_8_lut(blue5));
 	}
```

Comment:
The red6, green5, and blue5 are currently `int`, but the LUT function takes in uin8t_t: `inline uint8_t rgb5_to_8_lut(const uint8_t c)`

So suggest casting them inbound to the function.

the LUT function returns a uint8_t, which matches these functions' return types - so doesn't need casting.

```c++
return rgb5_to_8_lut(static_cast<uint8_t>(blue5));
```

Reply:
Done (in rgb555.h as well).